### DISCUSSION TITLE
Can the Isar db be altered externally?

### DISCUSSION THREAD

**Discussion by sunilguptasg** [isar/isar#1012] [Help]
Isar does not (and does not intend to) support encryption. I'm storing user data (essentially chat data) and this is not critical and need not be encrypted.

The question is can a rogue user alter the database, and make changes? 

How can I get notified of such changes/data corruption? In my case I can reload the data from a cloud dB. I just need to know that there have been changes made outside the app...

**Comment by JaffaKetchup** [comment]
@sunilguptasg The database can of course be edited manually, any file is vunerable to external corruption. If you try to open a corrupted database, you'll realise that the app likely fatally crashes without an explanation. If you implement a check system, you can isolate the faulty database and delete it before opening it.
